controllers/nodekeeper/nodekeeper_controller.go

```diff
@@ -121,7 +121,10 @@ func (r *ReconcileNodeKeeper) Reconcile(ctx context.Context, request reconcile.R
 		}
 		if hasFailed {
 			reqLogger.Info(fmt.Sprintf("Node drain timed out %s. Alerting.", node.Name))
-			metricsClient.UpdateMetricNodeDrainFailed(node.Name)
+			// Set metric only for the node going through upgrade
+			if r.Machinery.IsNodeUpgrading(node) {
```

Comment:
I am not quite sure if we add this condition check could fix the issue, as the existing logic, if we went to nodeDrain stage, the node should be under upgrade anyway.
Last time I saw the issue, it is not where we set the metric, but the metric was not unset after the node upgrade completed, even the node is gone, the alerts for that node is still firing.

Reply:
@bmeng There are multiple edge cases associated with this metric. In this case, we want to overcome the scenario where the drain strategy runs for the node that is marked unschedulable because the MHC detected node is unhealthy and deleted the machine object. Because the node is unschedulable as per MUO logic it detects the node is going through an upgrade, then runs a drain strategy for that node and sets this metric. MHC deletes the node and the metric never resets for the node. Therefore we added a logic to only set metrics for the node that is upgrading using **IsNodeUpgrading**